### DISCUSSION TITLE
Use cases for recurrence

### DISCUSSION THREAD

**Discussion by gour** [GothenburgBitFactory/taskwarrior#2447] [General]
Hello,

In order to not clutter #2338 further here I'm opening a new thread in regard to "recurrence overhaul"...

I still consider that TW is the best feature/ui-wise, but since I have lot of recurring tasks, adjusting time-shifts etc. has becoem too much work and seeing the further development kind of stalled I did move to [Etm](https://github.com/dagraham/etm-dgraham) which uses Python's [dateutil](https://dateutil.readthedocs.io/en/stable/) package and thanks to it it can define all kinds of complex recurring events.

Here is one example from the manual:

```
Text entry supports the full flexibility of the superb Python dateutil package. 
Consider, for example, creating a reminder for Presidential election day 
which repeats every 4 years on the first Tuesday after a Monday in November 
(a monthday falling between 2 and 8). In etm, this event would be

  * Presidential election day @s nov 1 2020 @r y &i 4 &M 11
    &m 2, 3, 4, 5, 6, 7, 8 &w tu
```

so I just wonder if it would be possible for TW-3.0 to take advantage of such libraries?

Sincerely,
Gour

**Comment by kwoot** [comment]
Incorporating a Python library into a C++ code base is a challenge on its own and not for the faint of heart.
Although I am not a C++ programmer I expect this to be marked as a future feature request at best.
Just my 2 cents of course.

**Comment by tbabej** [comment]
Indeed we would not be able to build upon the shoulders of dateutil, but we might be able to re-implement some of it behaviour. @gour Is there a specific behaviour that you have in mind?

**Comment by gour** [comment]
@tbabej Well, TW should be able, similar to Etm or org-mode support recurring tasks similar to the one I mentioned above. For the beginning, at least, it should be possible to have new instances of recurring tasks to be created after the task is *completed* and not based on recurring period. [Here](https://github.com/naggie/dstask/issues/12) is an interesting discussion in regard.
